certs_manager.cpp

```diff
@@ -84,6 +113,342 @@ void Manager::install(const std::string filePath)
+
+        if (CeLogin::CeLoginRc::Success == sRc)
+        {
+            // Verify system serial number is in machine list (and get the
+            // authorization)
+            const uint64_t serialNumberLengthParm = mSerialNumber.length();
+            sRc = CeLogin::decodeJson((const char*)sDecodedAsn->sourceFileData->data,
+                            sDecodedAsn->sourceFileData->length, mSerialNumber.c_str(),
+                            serialNumberLengthParm, *sJsonData);
+            if(CeLogin::CeLoginRc::Success == sRc ){
+                log<level::INFO>("Serial Number matches");
+            }
+            else{
+                log<level::ERR>("Serial Number does not match");
+            }
+        }
+    }
+    else if (CeLogin::CeLoginRc::SignatureNotValid == sRc)
+    {
+        log<level::ERR>("Signature is not valid");
+    }
+    else
+    {
+        log<level::ERR>("Error: ", entry("sRc=%d", (int)sRc));
+    }
+    //Check if time is expired
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        uint64_t sExpirationTime = 0;
+        std::time_t timeSinceUnixEpocInSecondsParm = std::time(NULL);
+        sRc = CeLogin::isTimeExpired( sJsonData, sExpirationTime, timeSinceUnixEpocInSecondsParm);
+        if(CeLogin::CeLoginRc::Success != sRc){
+            log<level::ERR>("ACF time expired");
+        }
+    }
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        //YYYYMMDD
+        char sTimeStr[20];
+        //int sLength =
+        sprintf(sTimeStr, "%04u-%02u-%02u", sJsonData->mExpirationDate.mYear,
+                    sJsonData->mExpirationDate.mMonth, sJsonData->mExpirationDate.mDay);
+        sDate = sTimeStr;
+    }
+
+    if (sDecodedAsn)
+        CELoginSequenceV1_free(sDecodedAsn);
+    if (sJsonData)
+        OPENSSL_free(sJsonData);
+    return sRc;
+}
+std::tuple<std::vector<uint8_t>, bool, std::string> Manager::getACFInfo(void){
+    bool isAcfInstalled = false;
+    bool prodKeyExists = false;
+    bool devKeyExists = false;
+
+    std::string sDate;
+    CeLogin::CeLoginRc sRc = CeLogin::CeLoginRc::Failure;
+    std::vector<uint8_t> accessControlFile{};
+    memset(accessControlFile.data(), 0, sizeof(accessControlFile)*accessControlFile.size());
+
+    try{
+        isAcfInstalled = std::filesystem::exists(ACF_FILE_PATH);
+        prodKeyExists = std::filesystem::exists(PROD_PUB_KEY_FILE_PATH);
+        devKeyExists = std::filesystem::exists(DEV_PUB_KEY_FILE_PATH);
+    }
+    catch(std::filesystem::filesystem_error& e) {
+            log<level::ERR>("filesystem_error error");
+    }
+
+    //ACF and production or development key should exist otherwise exit
+    if( !( (prodKeyExists || devKeyExists) && isAcfInstalled) ){
+        return std::make_tuple( accessControlFile, isAcfInstalled, sDate);
+    }
+
+    readBinaryFile(ACF_FILE_PATH, accessControlFile);
+    if(prodKeyExists){
+        std::vector<uint8_t> sPublicKeyFile;
+        readBinaryFile(PROD_PUB_KEY_FILE_PATH, sPublicKeyFile);
```

Comment:
return value readBinaryFie is not considered? what happens if it fail to open file?